airflow/models/taskinstance.py

```diff
@@ -1580,13 +1581,39 @@ def _coalesce_to_orm_ti(*, ti: TaskInstancePydantic | TaskInstance, session: Ses
 
 @internal_api_call
 @provide_session
-def _defer_task(
+def _defer_task_from_task_deferred(
     ti: TaskInstance | TaskInstancePydantic, exception: TaskDeferred, session: Session = NEW_SESSION
 ) -> TaskInstancePydantic | TaskInstance:
     from airflow.models.trigger import Trigger
 
     # First, make the trigger entry
     trigger_row = Trigger.from_object(exception.trigger)
+    updated_ti = _defer_task(
+        ti=ti,
+        session=session,
+        trigger_row=trigger_row,
+        trigger_kwargs=exception.kwargs,
+        next_method=exception.method_name,
+        timeout=exception.timeout,
+    )
+
+    session.merge(updated_ti)
+    session.commit()
+    return updated_ti
+
+
+@internal_api_call
+@provide_session
+def _defer_task(
```

Comment:
i think it's unnecessary to change the signature of this function and add the other two wrapper functions.

just build the exception object and pass it

```
e = TaskDeferred(trigger=BaseTrigger(...), method_name="...", ...)
```

unfortunately, we don't have integration tests set up for AIP-44 but if we did then it would reveal that this will break that mode of execution because for example there's no mechanism for serializing `Trigger`, possibly among other things.

obviously it's not your fault and a shortcoming of the current state, but in any case i think it will be simpler anyway to keep the existing signature and just build and exc object that you don't raise.

Reply:
What if we made `_defer_task` like this:

```
def `_defer_task(*, exception=None, start_trigger_args=None):
```
(omitting some kwargs)

Basically, let it accept either an exception or start_trigger_args, and proceed from there. That'd also should let us have 1 place to actually create the trigger record vs having multiple helper methods.

Reply:
Thanks for all your suggestions! I just refactored it a bit and everything looks much better!